src/views/WorkoutView.tsx

```diff
@@ -0,0 +1,268 @@
+import { useContext, useEffect } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { TimerContext, displayTimerDetails, totalWorkoutTime } from '../context/TimerContext';
+import type { Timer, TimerPhase } from '../types/timers';
+import { convertToMs } from '../utils/helpers';
+
+import { Button } from '../components/generic/Button';
+import { Panel } from '../components/generic/Panel';
+
+import TimerDisplay from '../components/timers/TimerDisplay';
+import { usePersistedState } from '../hooks/usePersistedState';
+import { useWorkoutHistory } from '../hooks/useWorkoutHistory';
+
+interface WorkoutState {
+    timers: Timer[];
+    currentTimerIndex: number;
+    running: boolean;
+    timeInMs: number;
+    currentPhase: TimerPhase;
+    currentRound: number;
+}
+
+const initialWorkoutState: WorkoutState = {
+    timers: [],
+    currentTimerIndex: 0,
+    running: false,
+    timeInMs: 0,
+    currentPhase: 'Work',
+    currentRound: 1,
+};
+
+const WorkoutView = () => {
+    const navigate = useNavigate();
+    const [searchParams] = useSearchParams();
+    const timersParam = searchParams.get('timers');
```

Comment:
This generally works but the url state is emptied when you navigate to a different page, like history or documentation. If I go back to the workout page, there are no url params. Which works okay, unless I refresh.

You could consider persisting the url state even when navigating between pages